### ISSUE TITLE
Calibration inflates UK population from 69M to 74M (should be ~70M)

### ISSUE THREAD

**Issue by MaxGhenis** [PolicyEngine/policyengine-uk-data#217]
## Summary

The dataset calibration process is inflating the UK population significantly above ONS targets. The base FRS 2025 dataset has ~69M people, but after calibration this jumps to ~74M - about 6% above the ONS mid-2024 actual estimate of 69.3M.

## Evidence

| Dataset | 2025 Population |
|---------|-----------------|
| Base dataset (frs_2025_with_ss.h5) | **68.97M** |
| Calibrated dataset (frs_2025_calibrated_v3.h5) | **73.57M** |
| ONS mid-2024 actual estimate | **69.3M** |
| ONS 2022-based projection for 2025 | **~70M** |

## Root Cause Investigation

1. The `uk_population` target IS included in the calibration loss function (`loss.py` line 318)
2. The population index in policyengine-uk (`ons.population`) uses reasonable growth rates
3. BUT the calibration is not constraining population properly - other targets are pulling weights in a direction that inflates total population

## Potential Solutions

1. **Increase the weight on population targets** in the calibration loss function
2. **Add a hard constraint** that total population must match the target
3. **Review conflicting targets** that may be inflating population (e.g., regional age bands sum to more than national total)

## Impact

This is causing CI test failures in PR #216:
- Test expected: 69.5M (ONS 2022-based)
- Test got: 73.7M

## Data Sources

- [ONS mid-2024 population estimate](https://www.ons.gov.uk/peoplepopulationandcommunity/populationandmigration/populationestimates/bulletins/annualmidyearpopulationestimates/mid2024): 69.3M
- [ONS 2022-based population projections](https://www.ons.gov.uk/peoplepopulationandcommunity/populationandmigration/populationprojections/bulletins/nationalpopulationprojections/latest)

**Comment by MaxGhenis** [comment]
## Root Cause Found

The calibration loss function gives **equal weight to all 556 targets**. Population is just 1/556 of the total loss, so it gets overridden by other targets.

**Targets breakdown:**
- Total targets: **556**
- Population target: `ons/uk_population` = 69.90M
- Largest targets: employment_income 1,133bn, income_tax 328bn, etc.

The loss function uses symmetric relative error:
```python
sre = ((1 + x) / (1 + y) - 1) ** 2
return mse_local + mse_national  # all targets weighted equally
```

**Solution Options:**

1. **Add population weight multiplier** in calibrate.py:
   ```python
   population_loss = sre(pred_pop, y_pop)
   return mse_local + mse_national + 10 * population_loss  # Give 10x weight
   ```

2. **Hard constraint**: Rescale all weights after optimization so total population matches exactly:
   ```python
   final_weights *= target_pop / actual_pop
   ```

3. **Target-specific weights**: Allow specifying weights for "anchor" targets like population

**Comment by MaxGhenis** [comment]
Closing as fixed/guarded by the merged calibration-fidelity work. PR #366 added population regression coverage after the April pipeline fixes (#359, #362, #363) brought the weighted UK population back within the ONS target tolerance; current CI on #414 also passed the population tests (`test_population.py` and `test_population_fidelity.py`). The original 73-74M overshoot is now guarded by the <72M regression test and the 4% ONS tolerance.
